internal/status/reporter.go

```diff
@@ -3,23 +3,38 @@ package status
 import (
 	"context"
 
+	"github.com/kubeshop/botkube/internal/loggerx"
+	"github.com/kubeshop/botkube/pkg/config"
+
 	"github.com/sirupsen/logrus"
 )
 
 type StatusReporter interface {
+	ReportDeploymentConnectionInit(ctx context.Context, k8sVer string) error
 	ReportDeploymentStartup(ctx context.Context) error
 	ReportDeploymentShutdown(ctx context.Context) error
 	ReportDeploymentFailure(ctx context.Context, errMsg string) error
 	SetResourceVersion(resourceVersion int)
+	SetLogger(logger logrus.FieldLogger)
 }
 
-func GetReporter(remoteCfgEnabled bool, logger logrus.FieldLogger, gql GraphQLClient, resVerClient ResVerClient, cfgVersion int) StatusReporter {
+func GetReporter(remoteCfgEnabled bool, gql GraphQLClient, resVerClient ResVerClient, logger *logrus.FieldLogger) StatusReporter {
```

Comment:
the `logrus.FieldLogger` it already an interface definition, so you should use a pointer. With Go is also better to avoid `if else` statements, and just return early, so I would propose sth like:
```go
func GetReporter(remoteCfgEnabled bool, gql GraphQLClient, resVerClient ResVerClient, log logrus.FieldLogger) StatusReporter {
	if remoteCfgEnabled {
		log  = withDefaultLogger(log)
		return newGraphQLStatusReporter(
			log.WithField("component", "GraphQLStatusReporter"),
			gql,
			resVerClient,
		)
	}

	return newNoopStatusReporter()
}

func withDefaultLogger(log logrus.FieldLogger) logrus.FieldLogger {
	if log != nil {
		return log
	}
	return loggerx.New(config.Logger{
		Level:         "info",
	})
}
```

or 
```go
func GetReporter(remoteCfgEnabled bool, gql GraphQLClient, resVerClient ResVerClient, log logrus.FieldLogger) StatusReporter {
	if remoteCfgEnabled {
		if log == nil {
			log = loggerx.New(config.Logger{
				Level: "info",
			})
		}

		return newGraphQLStatusReporter(
			log.WithField("component", "GraphQLStatusReporter"),
			gql,
			resVerClient,
		)
	}

	return newNoopStatusReporter()
}
```

up to you.

Reply:
Got you really nice